extensions/lifecycle/LifecycleConfigValidator.js

```diff
@@ -67,6 +67,21 @@ const joiSchema = joi.object({
     coldStorageRestoreAdjustTopicPrefix: joi.string().default('cold-restore-adjust-req-'),
     coldStorageGCTopicPrefix: joi.string().default('cold-gc-req-'),
     coldStorageStatusTopicPrefix: joi.string().default('cold-status-'),
+    supportedLifecycleRules: Joi.array().items(
+        Joi.string().valid(
+            'expiration',
+            'noncurrentVersionExpiration',
+            'abortIncompleteMultipartUpload',
+            'transitions',
+            'noncurrentVersionTransition'
+        )
+    ).default([
+        'expiration',
+        'noncurrentVersionExpiration',
+        'abortIncompleteMultipartUpload',
+        'transitions',
+        'noncurrentVersionTransition',
+    ]),
```

Comment:
Could this array be a constant in a single place?
Something like this (not tested)

```suggestion
        Joi.string().valid(...supportedLifecycleRules)
    ).default(supportedLifecycleRules),
```